File: lib/data/paper_setup.dart
import 'package:tsc_printer/constant/enum_consts.dart';

class PaperSetup {
  final int width;
  final int height;
  final int speed;
  final int density;
  final int sensor;
  final int sensorDistance;
  final int sensorOffset;

  PaperSetup({
    this.width = 50,
    this.height = 30,
    this.speed = 4,
    this.density = 8,
    this.sensor = 0,
    this.sensorDistance = 0,
    this.sensorOffset = 0,
  });

  factory PaperSetup.fromJson(Map<String, dynamic> json) {
    return PaperSetup(
      width: json['width'] as int? ?? 50,
      height: json['height'] as int? ?? 30,
      speed: json['speed'] as int? ?? 4,
      density: json['density'] as int? ?? 8,
      sensor: json['sensor'] as int? ?? 0,
      sensorDistance: json['sensorDistance'] as int? ?? 0,
      sensorOffset: json['sensorOffset'] as int? ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'width': width,
      'height': height,
      'speed': speed,
      'density': density,
      'sensor': sensor,
      'sensorDistance': sensorDistance,
      'sensorOffset': sensorOffset,
    };
  }
}

File: android/src/main/java/szk/kawanlama/tsc_printer/model/PaperSetup.java
package szk.kawanlama.tsc_printer.model;

import java.util.Map;
import java.util.HashMap;

public class PaperSetup {
    public int width = 50;
    public int height = 30;
    public int speed = 4;
    public int density = 8;
    public int sensor = 0;
    public int sensorDistance = 0;
    public int sensorOffset = 0;

    public static PaperSetup fromJson(Map<String, Object> json) {
        if (json == null) return null;
        PaperSetup setup = new PaperSetup();
        
        try {
            if (json.get("width") != null) setup.width = Integer.parseInt(String.valueOf(json.get("width")));
            if (json.get("height") != null) setup.height = Integer.parseInt(String.valueOf(json.get("height")));
            if (json.get("speed") != null) setup.speed = Integer.parseInt(String.valueOf(json.get("speed")));
            if (json.get("density") != null) setup.density = Integer.parseInt(String.valueOf(json.get("density")));
            if (json.get("sensor") != null) setup.sensor = Integer.parseInt(String.valueOf(json.get("sensor")));
            if (json.get("sensorDistance") != null) setup.sensorDistance = Integer.parseInt(String.valueOf(json.get("sensorDistance")));
            if (json.get("sensorOffset") != null) setup.sensorOffset = Integer.parseInt(String.valueOf(json.get("sensorOffset")));
        } catch (Exception e) {
            e.printStackTrace();
        }
        
        return setup;
    }

    public Map<String, Object> toJson() {
        Map<String, Object> json = new HashMap<>();
        json.put("width", width);
        json.put("height", height);
        json.put("speed", speed);
        json.put("density", density);
        json.put("sensor", sensor);
        json.put("sensorDistance", sensorDistance);
        json.put("sensorOffset", sensorOffset);
        return json;
    }
}
